src/cdomains/intDomain.ml

```diff
@@ -2307,16 +2307,40 @@ struct
     if M.tracing then  M.trace "congruence" "shift_left : %a %a becomes %a \n" pretty x pretty y pretty res;
     res
 
+  let find_power_of_two n =
+    let two = Ints_t.of_int 2 in
+    let rec find n acc =
+      if n %: two =: Ints_t.zero then
+        find (n /: two) (two *: acc)
+      else
+        acc
+    in
+    find n Ints_t.one
```

Comment:
What is this really computing? Finding the number of trailing zero bits?
If so, then this could be rather inefficient since there are CPU instructions for this sort of thing and `Z.trailing_zeros` could be a much more efficient way to compute that without all the multiplications and divisions.

Reply:
It computes the largest power of two that n is divisible by. Now this is computed with `Z.trailing_zeros` (as you suggest) and `shift_left`.